builtin/mainmenu/common.lua

```diff
@@ -146,35 +146,15 @@ end
 
 --------------------------------------------------------------------------------
 os.tempfolder = function()
-	if core.settings:get("TMPFolder") then
-		return core.settings:get("TMPFolder") .. DIR_DELIM .. "MT_" .. math.random(0,10000)
-	end
-
-	local filetocheck = os.tmpname()
-	os.remove(filetocheck)
-
-	-- luacheck: ignore
-	-- https://blogs.msdn.microsoft.com/vcblog/2014/06/18/c-runtime-crt-features-fixes-and-breaking-changes-in-visual-studio-14-ctp1/
-	--   The C runtime (CRT) function called by os.tmpname is tmpnam.
-	--   Microsofts tmpnam implementation in older CRT / MSVC releases is defective.
-	--   tmpnam return values starting with a backslash characterize this behavior.
-	-- https://sourceforge.net/p/mingw-w64/bugs/555/
-	--   MinGW tmpnam implementation is forwarded to the CRT directly.
-	-- https://sourceforge.net/p/mingw-w64/discussion/723797/thread/55520785/
-	--   MinGW links to an older CRT release (msvcrt.dll).
-	--   Due to legal concerns MinGW will never use a newer CRT.
-	--
-	--   Make use of TEMP to compose the temporary filename if an old
-	--   style tmpnam return value is detected.
-	if filetocheck:sub(1, 1) == "\\" then
-		local tempfolder = os.getenv("TEMP")
-		return tempfolder .. filetocheck
-	end
+	local temp = core.get_temp_path()
+	return temp .. DIR_DELIM .. "MT_" .. math.random(0, 10000)
+end
 
-	local randname = "MTTempModFolder_" .. math.random(0,10000)
-	local backstring = filetocheck:reverse()
-	return filetocheck:sub(0, filetocheck:len() - backstring:find(DIR_DELIM) + 1) ..
-		randname
+--------------------------------------------------------------------------------
+os.tmpname = function()
```

Comment:
Minetest doesn't actually use `os.tmpname` or `os.tmpfile`, so perhaps it's best not to override them

Reply:
This is only in the mainmenu code so it's not like there are compatibility concerns.